# try:
#     <потенциально опасный блок>

# except <тип исключения>:
#     <блок выполняемый при перехвате исключения>

# except <тип исключения 2>:
#     <блок выполняемый при перехвате исключения 2>

# except <тип исключения 3>:
#     <блок выполняемый при перехвате исключения 3>

# ...

# else:
#     <блок выполняемый при отсутсвии исключений>

# finally:
#     <блок выполняемый при любом исходе>

# try — обязательный блок + как минимум один блок except, finally или else
# except может быть не один

n1, n2 = int(input('>')), int(input('>'))


try:
    res = n1 / n2
except NameError:
    print('переменные не объявлены')
except ZeroDivisionError:
    print('деление на ноль невозможно')
else:
    print(f'{res = }')
finally:
    print('это print из блока finally')


print('print после try')